File: app/src/main/java/com/tkdev/weatherapp/utils/RetrofitCalls.java
package com.tkdev.weatherapp.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.view.View;

import com.google.android.material.snackbar.Snackbar;

import java.text.SimpleDateFormat;

public class RetrofitCalls {




    public static final String AND_SYMBOL = "&";
    public static final String WEATHER_REQUEST_BASE = "https://api.openweathermap.org/data/2.5/";
    public static final String WEATHER_CURRENT_REQUEST = "weather?";
    public static final String WEATHER_FORECAST_REQUEST = "forecast?";
    public static final String WEATHER_CITY_ID_PREFIX = "id";
    public static final String WEATHER_CITY_NAME = "q";
    public static final String WEATHER_CITY_ID = "2643743";
    public static final String WEATHER_CITY_ID_RETROFIT = "{city}";
    public static final String WEATHER_API_PREFIX = "APPID=";
    public static final String WEATHER_API_KEY = "5b08b54ce198509d241991110864cab4";
    public static final String WEATHER_TEMPERATURE_PREFIX = "units=";
    public static final String WEATHER_TEMPERATURE_CELSIUS = "metric";

    public static final String CELSIUS_SYMBOL = "°";
    public static final String HUMIDITY_SYMBOL = " %";
    public static final String FORECAST_DAY_PATTERN = "EEE";
    public static final String FORECAST_HOUR_PATTERN = "HH:mm";
    public static final String LAST_UPDATE_PATTERN = "HH:mm";
    public static final String DATE_PATTERN = "EEE dd-MM-yyyy";


    public static String temperaturePrefix(double temp) {
        return (Math.rint(temp * 10) / 10) + CELSIUS_SYMBOL;
    }

    public static String datePattern(Integer dt, String pattern) {
        SimpleDateFormat dateFormat = new SimpleDateFormat(pattern);
        Long date = Long.valueOf(dt) * 1000;
        return dateFormat.format(date);
    }

    public static void makeSnack(View view, String message){
        Snackbar.make(view, message, Snackbar.LENGTH_SHORT).show();
    }

}

File: app/src/main/java/com/tkdev/weatherapp/common/domain/RetrofitCalls.kt
package com.tkdev.weatherapp.common.domain

import android.view.View
import androidx.annotation.VisibleForTesting
import com.google.android.material.snackbar.Snackbar
import java.text.SimpleDateFormat
import kotlin.math.round

class RetrofitCalls {
    companion object {
        const val AND_SYMBOL = "&"
        const val WEATHER_REQUEST_BASE = "https://api.openweathermap.org/data/2.5/"
        const val WEATHER_CURRENT_REQUEST = "weather?"
        const val WEATHER_FORECAST_REQUEST = "forecast?"
        const val WEATHER_CITY_ID_PREFIX = "id"
        const val WEATHER_CITY_NAME = "q"
        const val WEATHER_CITY_ID = "2643743"
        const val WEATHER_CITY_ID_RETROFIT = "{city}"
        const val WEATHER_API_PREFIX = "APPID="
        const val WEATHER_API_KEY = "5b08b54ce198509d241991110864cab4"
        const val WEATHER_TEMPERATURE_PREFIX = "units"
        const val WEATHER_TEMPERATURE_CELSIUS = "metric"
        const val WEATHER_TEMPERATURE_FARHENHEIT = "imperial"

        const val CELSIUS_SYMBOL = "°"
        const val KELVIN_SYMBOL = "°K"
        const val FARHENHEIT_SYMBOL = "°F"
        const val HUMIDITY_SYMBOL = " %"
        const val FORECAST_DAY_PATTERN = "EEE"
        const val FORECAST_HOUR_PATTERN = "HH:mm"
        const val LAST_UPDATE_PATTERN = "HH:mm"
        const val DATE_PATTERN = "EEE dd-MM-yyyy"


        fun temperaturePrefix(temp: Double, prefix: String): String{
            var newTempPref: String = (round(temp * 10) / 10).toString()
            newTempPref += when(prefix){
                WEATHER_TEMPERATURE_CELSIUS -> CELSIUS_SYMBOL
                WEATHER_TEMPERATURE_FARHENHEIT -> FARHENHEIT_SYMBOL
                else -> KELVIN_SYMBOL
            }
            return newTempPref
        }

        fun humidityPrefix(humidity: Int): String = humidity.toString() + HUMIDITY_SYMBOL

        fun datePattern(dt: Long, pattern: String): String
            = SimpleDateFormat(pattern).format(dt)
    }
}